Looks like a compiler error on Windows:

```
11:58:21 [7.265s]   has_resource.cpp
11:58:21 [7.984s] C:\J\workspace\ci_windows\ws\src\ament\ament_index\ament_index_cpp\src\get_resources.cpp(75,58): error C2440: 'initializing': cannot convert from 'rcpputils::fs::path' to 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' [C:\J\workspace\ci_windows\ws\build\ament_index_cpp\ament_index_cpp.vcxproj]
11:58:21 [8.578s] C:\J\workspace\ci_windows\ws\src\ament\ament_index\ament_index_cpp\src\get_resources.cpp(75,58): message : No constructor could take the source type, or constructor overload resolution was ambiguous [C:\J\workspace\ci_windows\ws\build\ament_index_cpp\ament_index_cpp.vcxproj]
```

--- a/ament_index_cpp/src/get_resource.cpp
+++ b/ament_index_cpp/src/get_resource.cpp
@@ -42,7 +42,7 @@ get_resource(
   auto paths = get_search_paths();
   for (auto path : paths) {
     auto resource_path = rcpputils::fs::path{path} / "share" / "ament_index" / "resource_index" /
-      resource_type / resource_name;
+    resource_type / resource_name;
     std::ifstream s(resource_path.string());
     if (s.is_open()) {
       std::stringstream buffer;

--- a/ament_index_cpp/src/get_resources.cpp
+++ b/ament_index_cpp/src/get_resources.cpp
@@ -72,7 +72,7 @@ get_resources(const std::string & resource_type)
     closedir(dir);
 
 #else
-    std::string pattern = rcpputils::fs::path{path} / "*";
+    const auto pattern = (path / "*").string();
     WIN32_FIND_DATA find_data;
     HANDLE find_handle = FindFirstFile(pattern.c_str(), &find_data);
     if (find_handle == INVALID_HANDLE_VALUE) {

--- a/ament_index_cpp/src/has_resource.cpp
+++ b/ament_index_cpp/src/has_resource.cpp
@@ -40,7 +40,7 @@ has_resource(
   auto paths = get_search_paths();
   for (auto path : paths) {
     auto resource_path = rcpputils::fs::path{path} / "share" / "ament_index" / "resource_index" /
-      resource_type / resource_name;
+    resource_type / resource_name;
     std::ifstream s(resource_path.string());
     if (s.is_open()) {
       if (prefix_path) {

--- a/ament_index_cpp/test/utest.cpp
+++ b/ament_index_cpp/test/utest.cpp
@@ -21,6 +21,8 @@
 #include <stdexcept>
 #include <string>
 
+#include "rcpputils/filesystem_helper.hpp"
+
 #include "ament_index_cpp/get_package_prefix.hpp"
 #include "ament_index_cpp/get_package_share_directory.hpp"
 #include "ament_index_cpp/get_packages_with_prefixes.hpp"
@@ -253,8 +255,10 @@ TEST(AmentIndexCpp, get_package_share_directory) {
   subfolders.push_back("prefix2");  // only contains bar and baz packages
   set_ament_prefix_path(subfolders);
   // bar is in both, but prefix 1 takes precedence
+  const auto subfolder_path =
+    rcpputils::fs::path{generate_subfolder_path("prefix1")} / "share" / "bar";
   EXPECT_EQ(
-    generate_subfolder_path("prefix1") + "/share/bar",
+    subfolder_path.string(),
     ament_index_cpp::get_package_share_directory("bar"));
 }
 

